src/System.Memory/tests/Performance/Perf.ReadOnlySequence.First.cs

```diff
@@ -0,0 +1,367 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Memory.Tests;
+using System.MemoryTests;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.Buffers.Tests
+{
+    public class Rerf_ReadOnlySequence_First
+    {
+        private const int InnerCount = 1000;
+        volatile static int _volatileInt = 0;
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000, 0, 1000)]
+        [InlineData(1000, 100, 800)]
+        [InlineData(1000 * 1000, 0, 1000 * 1000)]
+        [InlineData(1000 * 1000, 1000, 998 * 1000)]
+        private static void Byte_Array(int size, int start, int lenght)
+        {
+            var buffer = new ReadOnlySequence<byte>(new byte[size], start, lenght);
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        for (int j = 0; j < size / 10; j++)
+                        {
+                            ReadOnlyMemory<byte> first = buffer.First;
+                            localInt ^= first.Length;
+                        }
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000, 0, 1000)]
+        [InlineData(1000, 100, 800)]
+        [InlineData(1000 * 1000, 0, 1000 * 1000)]
+        [InlineData(1000 * 1000, 1000, 998 * 1000)]
+        private static void Byte_Memory(int size, int start, int lenght)
```

Comment:
I don't think there is much utility in a performance test like this one. It should be identical to the Byte_Array test.

Reply:
It is test for ROS with MemoryManager inside.

Reply:
Is there any difference in performance between the Byte_Array and Byte_Memory tests?

Run the performance tests and share the results. In the System.Memory/tests/Performance directory, run this command:
`msbuild /t:BuildAndTest /p:Performance=true /p:ConfigurationGroup=Release /p:TargetOS=Windows_NT`

See https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/performance-tests.md#building-and-running-tests

Reply:
> Is there any difference in performance between the Byte_Array and Byte_Memory tests?

~ 2.5 times slow.